R/edgeThresh.R

```diff
@@ -0,0 +1,284 @@
+    # Read EGG
+    # Use the helper function ".permute" to create permutations,
+    # which takes a copy of EGG, and N, and returns N random
+    # graphs as a list (?) of objects
+
+
+    # If variables are assigned in the local environmet, the helper functions
+    # can use them.
+    # Generate the list of edges, vertices and vertex count available in EGG
+    allEdges <- igraph::get.edgelist(EGG)
+    allVertices <- igraph::V(EGG)
+    numVertices <- igraph::vcount(EGG)
+
+    # Call permute on EGG, store returned list of graphs as an object
+    graphList <- .permute(EGG, N)
+    masterDeltaList <- vector("list", length = N)
+
+    # For each graph:
+
+    # Create a vector storing min δ ord for each maxComponentSize
+    # (= is used to only store objects in current loop)
+    graphDeltas = vector("numeric", Lmax - 1)
+    currentDelta = 0.01
+    currentGraph = graphList[i]
+    currentEdges = igraph::E(currentGraph)
+
+    # Set maxComponentSize to 2 initially, and increase to a max of Lmax.
+    maxComponentSize = 2
+
+    # For each Lmax, set δ to 0.01 (?), remove all edges that have weights that
+    # are below such δ.
+    currentGraph <- igraph::delete.edges(currentGraph, which(currentEdges$weight <= currentDelta))
+
+    # Find the largest 'strongly' connected component:
+    largestComponent <- max(igraph::components(currentGraph, mode = "strong")$csize)
+
+    # If largest component has size less than Lmax,
+    # continue incrementing δ, else, stop and record δ in the vector, and
+    # increase maxComponentSize by 1 (to a max of Lmax)
+    if (largestComponent <= maxComponentSize) {
+        currentDelta <- currentDelta + 0.01
+    } else {
+        graphDeltas[i] <- currentDelta
+        maxComponentSize = maxComponentSize + 1
+        if (maxComponentSize > Lmax) {
+            # Must find another way to add to list if parallelization is chosen
+            masterDeltaList[[i]] <- graphDeltas
+            break
+        }
+    }
+    # (What if the first δ leads to a size equal to maxComponentSize? Should δ
+    # be decreased instead? By how much should δ be incremented at each step?)
+
+    # Find the minimum δ for each graph at each maxComponentSize
+
+    # After generating N vectors of size (Lmax - 1), find the median δ for each
+    # δ ord (maxComponentSize). i.e. Lmax - 1 median values.
+    deltaOrd <- vector("numeric", Lmax - 1)
+    for (i in 1:(Lmax - 1)) {
+        deltaOrd[i] <- median(sapply(masterDeltaList, FUN = "[", i))
+    }
+
+    # Attach the vector as graph metadata to the original EGG and return.
+    # (Is this the correct way?)
+    attributes(EGG) <- deltaOrd
+
+    # (Can this be parallelized? Since the processes are not interdependent)
+
+    # Write to Console:
+    # Update the fraction of N graphs that are done with permutations, (can
+    # also update the rough percentage of Q * |E| swaps that are done, but can
+    # slow down the process). This is done in .permute.
```

Comment:
I don't understand what you mean here. Console output is a general overview of the state of the process, not an exact record of data - unless you are requesting user input. In general, some progress indicator is good if time to completion takes longer than about five seconds.